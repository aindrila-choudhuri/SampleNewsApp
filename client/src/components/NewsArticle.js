import React from "react";

const NewsArticle = (props) => {
    const {news} = props

    const defaultImageURL = process.env.PUBLIC_URL + "/news.jpeg"

    return(
        <div data-testid="newsarticle" className="news" onClick={()=>window.location.href=news.url}>
            <img className="image__desc" src={news.urlToImage ? news.urlToImage : defaultImageURL} />
            <h4 className="news__title">{news.title}</h4>
            <span className="news__author">{news.author}</span> <br />
            <span className="news__published">{new Date(news.publishedAt).toString()}</span>
            <span className="news__source">{news.source.name}</span>
        </div>
    )
}

export default NewsArticle;