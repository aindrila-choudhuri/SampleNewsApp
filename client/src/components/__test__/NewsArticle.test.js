import React from 'react';
import ReactDOM from 'react-dom';
import NewsArticle from "./../NewsArticle";
import {render, cleanup} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';

const newsObj = {
    source: {
        "id": null,
        "name": "Yahoo Entertainment"
    },
    author: "AFP",
    title: "Vaccine hopes boost British pound",
    description: "The British pound continued a frantic two-month rise against the euro and the dollar, reaching new highs Monday in a sign of traders' enthusiasm for the...",
    url: "https://news.yahoo.com/vaccine-hopes-boost-british-pound-032804944.html",
    urlToImage: "https://s.yimg.com/uu/api/res/1.2/G2nZb5KC19af_vZPEDV7.g--~B/aD01MTI7dz03Njg7YXBwaWQ9eXRhY2h5b24-/https://media.zenfs.com/en/afp.com/0d2d90fb77d26ecf1317517a3e55fe84",
    publishedAt: "2021-02-23T03:28:04Z",
    content: "The British pound continued a frantic two-month rise against the euro and the dollar, reaching new highs Monday in a sign of traders' enthusiasm for the country's vaccination roll-out.\r\nJust before Câ€¦ [+2777 chars]"
};

afterEach(cleanup);

it("renders NewsArticle component without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<NewsArticle news={newsObj}></NewsArticle>, div);
});

it("renders NewsArticle component correctly", () => {
    const {getByTestId} = render(<NewsArticle news={newsObj}></NewsArticle>);
    const newsarticle = getByTestId('newsarticle');
    
    expect(newsarticle).toContainElement(newsarticle);
    expect(newsarticle).toHaveClass("news")
});

it("matches snapshot", () => {
    const domTree = renderer.create(<NewsArticle news={newsObj}></NewsArticle>).toJSON();
    expect(domTree).toMatchSnapshot();
});