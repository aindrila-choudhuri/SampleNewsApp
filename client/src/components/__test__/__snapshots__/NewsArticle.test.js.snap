// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="news"
  data-testid="newsarticle"
  onClick={[Function]}
>
  <img
    className="image__desc"
    src="https://s.yimg.com/uu/api/res/1.2/G2nZb5KC19af_vZPEDV7.g--~B/aD01MTI7dz03Njg7YXBwaWQ9eXRhY2h5b24-/https://media.zenfs.com/en/afp.com/0d2d90fb77d26ecf1317517a3e55fe84"
  />
  <h4
    className="news__title"
  >
    Vaccine hopes boost British pound
  </h4>
  <span
    className="news__author"
  >
    AFP
  </span>
   
  <br />
  <span
    className="news__published"
  >
    Tue Feb 23 2021 08:58:04 GMT+0530 (India Standard Time)
  </span>
  <span
    className="news__source"
  >
    Yahoo Entertainment
  </span>
</div>
`;
